handlers/handlers.go

```diff
@@ -0,0 +1,67 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/Daniorocket/cookit-backend/lib"
+	"github.com/gorilla/mux"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type Handler struct {
+	Client       *mongo.Client
+	DatabaseName string
+}
+type JWT struct {
+	Token    string
+	Username string
+}
+type Pagination struct {
+	Data          interface{} `json:"data"`
+	Limit         string      `json:"limit"`
+	Page          string      `json:"page"`
+	TotalElements int64       `json:"totalElements"`
+}
+type ApiResponse struct {
+	Data   interface{} `json:"data"`
+	Status string      `json:"status"`
+	Code   int         `json:"code"`
+	Error  string      `json:"error"`
+}
+
+func Authenticate(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch mux.CurrentRoute(r).GetName() {
+		case "CreateRecipe", "ListRecipes", "Renew", "CreateCategory", "GetUserinfo": //Todo better way? Handlers which require auth
```

Comment:
Z reguły łatwiej zaznaczyć które endpointy nie wymagają Auth'a w API niż które wymagają, jest ich mniej :D

Reply:
Racja, a można to zrobić w lepszy sposób? W którejś odpowiedzi o tym pisałem, ale chciałem, aby na podstawie nazwy (mux.CurrentRoute(r).GetName()) znalazło obiekt z tablicy routingu i sprawdziło jaką wartość ma pole NeedAuth. Obiekty są w pliku routing/routes.go. Zrobić coś w stylu funkcji, która zwraca tablicę routingu, czy to raczej przekombinowane i działać na nazwach, jak do tej pory, tylko oczywiście odwrotnie?

Reply:
Jeśli chcesz to dynamicznie z tych twoich struktur no to byś do Authenticate przekazać jeszcze liste (o widzę że jest Publiczna a raczej nie powinna być ale mniejsza :D)
https://github.com/Daniorocket/cookit-backend/pull/1/files/ce0db046767f9b66c6e2dc64af315bd46a656f8a#diff-f096cd6b7a0412bb34810438d4fd295f3acbc3c03dddfaf4fd7d60e293546572R17

i możesz przeforować po niej i sprawdzać zamaist switcha
```go
currentEndpoint := mux.CurrentRoute(r).GetName()
for _, endpoint := routing.Route {
    if currentEndpoint == endpoint {
        // i tutaj logika co ma się dziać 
    }
}